### PR TITLE
Adjust HTTP protocol to take Payloads not Payload.

### PR DESCRIPTION
**Pull request by robholland**
## What was changed
The Remote Encoding protocol is adjusted to take Payloads protobuf, not Payload.

## Why?
This makes for more efficient HTTP traffic.

### CODE REVIEW

**Comment by bergundy** [comment]
@robholland some messages use a single `Payload`, are we going to wrap those in `Payloads`?

**Comment by robholland** [comment]
Yes we will. You can now see an example of that here: https://github.com/temporalio/sdk-go/pull/733/files#diff-a4f99464f14541990ef49cd646b85168d2b4c3930138a5124d7d8adae6c68004R444

File: converter/encoding_data_converter_test.go [hunk 1]
```diff
@@ -205,30 +205,71 @@ func TestPayloadEncoderHTTPHandler(t *testing.T) {
...
 	require.Equal(t, http.StatusOK, rr.Code)
-	decodedPayloadJSON := strings.TrimSpace(rr.Body.String())
-	require.Equal(t, string(payloadJSON), decodedPayloadJSON)
+	decodedPayloadsJSON := strings.TrimSpace(rr.Body.String())
+	require.Equal(t, string(payloadsJSON), decodedPayloadsJSON)
+}
+
+func TestRemoteDataConverter(t *testing.T) {
+	defaultConv := converter.GetDefaultDataConverter()
+	encoder := converter.NewZlibEncoder(converter.ZlibEncoderOptions{AlwaysEncode: true})
```
Comment (cretz) [inline converter/encoding_data_converter_test.go:238]: Not necessarily needed _right now_, but ideally this test could have multiple encoders to test the loops are in the proper order.

File: converter/encoding_data_converter.go [hunk 2]
```diff
@@ -315,33 +320,142 @@ func (e *encoderHTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-type remotePayloadEncoder struct {
-	options RemotePayloadEncoderOptions
+// remoteDataConverter is a DataConverter that wraps an underlying data
+// converter and uses a remote encoder to handle encoding/decoding.
+type remoteDataConverter struct {
+	parent  DataConverter
+	options RemoteDataConverterOptions
+}
+
+var _ DataConverter = &remoteDataConverter{}
```
[the remark this answers is not in the document]
Comment (robholland) [reply converter/encoding_data_converter.go]: ah, thanks. Removed now.

File: converter/encoding_data_converter_test.go (hunk 1 above)
Comment (robholland) [reply converter/encoding_data_converter_test.go]: Adjusted the test to confirm this behaviour.

**Pull request merged**